### ISSUE TITLE
Save Roberta output features and reuse same to train linear classifier

### ISSUE THREAD

**Issue by suryapa1** [ThilinaRajapakse/simpletransformers#796]
**Describe the bug**

I have my roberta classifier model ready, As per my understanding. input_data==> roberta_tokenizer==> (input_id, attention)_id, segment_id)==> Pretrained_ropberta_model ==> Linear layer with num_labels. is flow of data right from input to final softmax output.

My goal is to do the following:

1) Shall I save the output of RobertBERTa as features as "bert_output" and reuse the same later point to train linear classifier alone in my incremental training
2) Once we have, let say examples count:100 collected in my feedback loop and train an only linear classifier with these examples without touching previous layers Roberta-pre trained weights.
3) Here I don't want to reuse cached features, since it is reversible vectors as per my requirement.

Any code snippet or help greatly appreciated.

**Comment by ThilinaRajapakse** [comment]
If I am understanding correctly, you want to train the whole RoBERTa model initially and then only the classification layer? You could do this in two stages.

1. Train the model normally.
2. Train the model with the `train_custom_parameters_only` option [here](https://simpletransformers.ai/docs/tips-and-tricks/#train-custom-parameters-only).

The flow of data you showed looks correct. This is the forward() method of a Roberta classification model.

```python
    def forward(
        self,
        input_ids=None,
        attention_mask=None,
        token_type_ids=None,
        position_ids=None,
        head_mask=None,
        inputs_embeds=None,
        labels=None,
    ):
        outputs = self.roberta(
            input_ids,
            attention_mask=attention_mask,
            token_type_ids=token_type_ids,
            position_ids=position_ids,
            head_mask=head_mask,
        )
        sequence_output = outputs[0]
        logits = self.classifier(sequence_output)

        outputs = (logits,) + outputs[2:]
        if labels is not None:
            if self.num_labels == 1:
                #  We are doing regression
                loss_fct = MSELoss()
                loss = loss_fct(logits.view(-1), labels.view(-1))
            else:
                loss_fct = CrossEntropyLoss(weight=self.weight)
                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
            outputs = (loss,) + outputs

        return outputs
```
